### ISSUE TITLE
"Deferred has no attribute send"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8861]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8861|
|Type|release blocker: regression|
|Created|2016-10-15 08:12:15Z|
|Branch|https://github.com/twisted/twisted/tree/8861-deferred-send|

Caused by [#8087](https://github.com/twisted/twisted/issues/8087), no idea why it isn't covered by the tests.

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/twisted/internet/defer.py", line 455, in callback
    self._startRunCallbacks(result)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/twisted/internet/defer.py", line 563, in _startRunCallbacks
    self._runCallbacks()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/twisted/internet/defer.py", line 649, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/twisted/internet/defer.py", line 1316, in gotResult
    _inlineCallbacks(r, g, deferred)
--- <exception caught here> ---
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/twisted/internet/defer.py", line 1260, in _inlineCallbacks
    result = g.send(result)
  File "t.py", line 8, in crawl
    results[page] = await treq.content(await treq.get(page))
builtins.AttributeError: 'Deferred' object has no attribute 'send'
```

<details><summary>Searchable metadata</summary>

```
trac-id__8861 8861
type__release_blocker__regression release blocker: regression
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8861_deferred_send 8861-deferred-send
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1476519135357759 1476519135357759
changetime__1476946309055111 1476946309055111
version__None None
owner__Amber_Brown__hawkowl_____ Amber Brown <hawkowl@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Tests running, PR at https://github.com/twisted/twisted/pull/563/files

**Comment by twisted-trac** [comment]
|<img alt="meejah's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|meejah commented|
|-|-|

In some tests I was working with, I needed to `await d` on a `Deferred` that had not yet `.callback()`'d to see the exception. I'll see if I can make a test for this fail properly in Twisted.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_4"></a>|@markrwilliams commented|
|-|-|

https://github.com/twisted/twisted/pull/563#pullrequestreview-4380033

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset c2631e266ac0ce4bda0cb19899773934a32779bb
```
#!CommitTicketReference repository="" revision="c2631e266ac0ce4bda0cb19899773934a32779bb"
Merge 8861-deferred-send: Add .send to Deferred for async/await

Author: hawkowl
Reviewer: markrwilliams
Fixes: #8861
```
